### DISCUSSION TITLE
Transactions and rollback

### DISCUSSION THREAD

**Discussion by fullben** [JanWiemer/jacis#19] [Q&A] [answered]
As far as I understand it, there are two general ways for running transactions:

```java
// Option 1
...
JacisLocalTransaction tx = container.beginLocalTransaction();
// Do work
tx.commit();

...

// Option 2
container.withLocalTx(() -> {...});
```
If I'm not mistaken, using option 1 would not result in a rollback being executed if an exception occurs in my block of work, while a rollback would be attempted when using option 2 and exception occurs.
Assuming this is the case, wouldn't it make sense to offer something like ` public <T> T withLocalTx(Supplier<T> task)` alongside the existing ` public void withLocalTx(Runnable task)`? I often find myself in lengthy transactions that in the end produce some value which I want to return.
Not sure if the above is a feature request or if I'm just missing something...

**Comment by JanWiemer** [comment, marked as answer]
Again a good idea for an extension! See issue: https://github.com/JanWiemer/jacis/issues/25#issue-986717281
